examples/cpp/flow_control/SubscriberApp.cpp

```diff
@@ -75,7 +78,9 @@ SubscriberApp::SubscriberApp(
     }
 
     // Create DataReader
-    reader_ = subscriber_->create_datareader(topic_, DATAREADER_QOS_DEFAULT, this, StatusMask::all());
+    DataReaderQos reader_qos = DATAREADER_QOS_DEFAULT;
+    subscriber_->get_default_datareader_qos(reader_qos);
+    reader_ = subscriber_->create_datareader(topic_, reader_qos, this, StatusMask::all());
```

Comment:
Since the participant is already has a listener, we don't need it here.
```suggestion
    reader_ = subscriber_->create_datareader(topic_, reader_qos);
```